src/models/tabularmodel.jl

```diff
@@ -0,0 +1,129 @@
+function emb_sz_rule(n_cat)
+    min(600, round(1.6 * n_cat^0.56))
+end
+
+function _one_emb_sz(cardinalitydict, catcol, sz_dict=nothing)
+    sz_dict = isnothing(sz_dict) ? Dict() : sz_dict
+    n_cat = cardinalitydict[catcol]
+    sz = catcol in keys(sz_dict) ? sz_dict[catcol] : emb_sz_rule(n_cat)
+    Int64(n_cat)+1, Int64(sz)
+end
+
+"""
+    get_emb_sz(cardinalitydict, cols; sz_dict=nothing)
+
+Returns a collection of tuples containing embedding dimensions for each column 
+in `cols`. `cardinalitydict` is an indexable collection which maps to the cardinality
+for each column present in `cols`. 
+
+## Keyword arguments
+
+- `sz_dict`: An indexable collection which may contain the required embedding 
+    size for a particular column present in `cols`. If not passed, then the size is
+    calculated using fastai's rule of thumb for embedding dimensions.
+"""
+
+function get_emb_sz(cardinalitydict, cols; sz_dict=nothing)
+    map(catcol -> _one_emb_sz(cardinalitydict, catcol, sz_dict), cols)
+end
+
+function sigmoidrange(x, low, high)
+    @. Flux.sigmoid(x) * (high - low) + low
+end
+
+function tabular_embedding_backbone(embedding_sizes, dropoutprob=0.)
+    embedslist = [Flux.Embedding(ni, nf) for (ni, nf) in embedding_sizes]
+    emb_drop = dropoutprob==0. ? identity : Dropout(dropoutprob)
+    Chain(
+        x -> tuple(eachrow(x)...), 
+        Parallel(vcat, embedslist), 
+        emb_drop
+    )
+end
+
+function tabular_continuous_backbone(n_cont)
+    BatchNorm(n_cont)
+end
+
+"""
+    TabularModel(catbackbone, contbackbone, [finalclassifier]; kwargs...)
+    TabularModel(catcols, `n_cont::Number, outsz::Number[; kwargs...)
```

Comment:
```suggestion
    TabularModel(catcols, n_cont::Number, outsz::Number; kwargs...])
```

Reply:
Is that final `]` necessary either? Never seen that syntax for optional args before.